crawl.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime, timedelta
+from os import makedirs, path
+
+import pandas as pd
+import requests
+
+import reddit_auth
+
+TOKEN = reddit_auth.get_token()
+
+def fetch_posts(number:int = 3, page:str = None):
+  """fetches # of posts, starting at page reference"""
+
+  params = {'limit': number}
+  if page:
+    params['after'] = page
+
+  headers = {
+    'User-Agent': 'ProductsFromReddit/0.0.1',
+    'Authorization': f'bearer {TOKEN}'
+  }
+  res = requests.get(
+    'https://oauth.reddit.com/r/hardwareswap/new',
+    headers = headers,
+    params = params
+  )
+
+  # Dump response data posts into dataframe for manipulation
+  posts = pd.json_normalize(res.json()['data']['children'])
+  posts.columns = posts.columns.str.replace('data.', '', regex=True)
+
+  next_page = res.json()['data']['after']
+
+  return posts, next_page
+
+
+
+def save_posts_to_json(posts):
+  "writes posts to separate json files in data/ directory"
```

Comment:
[Nit] Docstrings on functions use the triple quote string format and should typically be capitalized.

```suggestion
  "Writes posts to separate json files in data/ directory."
```